README.md

```diff
@@ -20,153 +20,82 @@ This contract is intended to operate as the distributor, and has functionality t
 
 ### The watchlist
 
-The injector runs using a watch list.  The watch list is defined as the tuple of [gaugeAddress, amount, maxTopups].
+The injector runs using a watch list. The watch list is defined as the tuple of [gaugeAddress, amountPerPeriod, maxPeriods, doNotStartBeforeTimestamp].
 
-For every streamer address, assuming a sufficent token balance, the injector will inject the specified amounts each epoch until it has done so maxTopups time.
+For every streamer address, assuming a sufficent token balance, the injector will inject the specified amounts each epoch until it has done so maxPeriods time.
 
-List changes are atomic.  A new list replaces an old one.
+It's possible to add new recipients and configurations without altering or including the old ones (difference to v1).
 
-This list is defined by calling the function `setRecipientList(streamerAddresses, amountsPerPeriod, maxPeriods)` on the deployed injector.
-
-`setValidatedRecipientList(streamerAddresses, amountsPerPeriod, maxPeriods)` can also be called.  It checks that there are not running programs that still have periods to pay out, and that the balance in the injector contract is sufficient to pay the full new program. It will revert if these conditions are not met, and set a new list if they are. 
+This list is defined by calling the function `addRecipients(streamerAddresses, amountsPerPeriod, maxPeriods, doNotStartBeforeTimestamp)` on the deployed injector.
 
 
 ### Balances
 The injector uses ERC20 balances in the injector contract to pay rewards.  The upkeeps will not run if there is not enough tokens in the contract to satisfy all currently due injections.
 
 The following usage pattern can be followed to maintain proper balances at all times:
 
-Note: You can use `setRecipientList` to bypass the validation and set a program without checking balances or if a running program is being overwritten
-
 #### When setting schedule
-- Ensure that there is 0 token balance in the contract
 - Transfer the exact amount required for the entire program (all streams, all amounts, all periods)
-- Use `setValidatedRecipientList(streamerAddresses, amountsPerPeriod, maxPeriods)`
-- Do not load the next schedule until the current one is complete.
+- Use `addRecipients(streamerAddresses, amountsPerPeriod, maxPeriods, doNotStartBeforeTimestamp)`
 
 
 #### To abort a schedule midway through or reset
-- Use `setRecipientList([], [], [])` to clear the list.
+- Use `removeRecipients(recipients)` to one or multiple configurations from the list.
 - Use `sweep(token)` to transfer any remaining tokens back to the owner.
 - Now you can use the normal process to set a new schedule.
 
 ## Deployment and operations
 
 ### Dependancies/environment setup
```

Comment:
Where can a user find information about how to setup all the proper scanner tokens in hardhat for automatic verification of smart contracts?  How can one configure rpcs?  Assume that someone using this is has no hardhat experience and needs some tips about how to set it up to deploy across all of our chains.